### DISCUSSION TITLE
Supbase query works, but API call returns empty result

### DISCUSSION THREAD

**Discussion by sanjaychristian** [supabase/supabase#28995] [Questions] [answered]
I have a table in supbase with the following RLS:
alter policy "Enable select for authenticated users only" on "public"."userCampaign_txt" to authenticated using (  ("ownerId" = auth.uid()));
Scenario 1: Works; When I call using "Supabase Query", I get the result.
Scenario 2: Does Not Work: I am using API call for the same table and getting empty result: During my api call, I am passing JWT token in "Authorization" header.  my auth.uid() is null.  I can see the JWT information passing through properly (status code: 200 ok).
Has anyone encountered this issue?  Please help.  Thanks.

**Comment by gc-ft** [comment]
We need more context in order to help, looks like you are not setting the Authorization header correctly. Can you paste some code that actually sets up the API and calls it?

Also, when you say "Supabase Query" you mean the supabase javascript (or some other language) library? And you want to use REST API directly? Or what are you actually meaning by the two Scenarios?

**Comment by sanjaychristian** [comment]
Thank you for looking into this.  Here is the detail regarding your ask.
[Flutterflow  authentication upon “Login” button.pdf](https://github.com/user-attachments/files/16819660/Flutterflow.authentication.upon.Login.button.pdf)

**Reply by gc-ft** [reply]
Hi,

Maybe someone else can decipher that has experience with flutterflow, however without actual details about what exactly you are doing this seems very hard to debug this way.

From the request information you have sent there the first way that works is a using the actual supabase API, the second way is using the supabase API via an rpc call, so a postgres function that was created. There is a heroku url in front your endpoint which I am not sure is an issue that is causing the problem or "normal" with flutterflow.

The debug screenshots btw do not match the queries you are showing...

Anyway, i believe only someone with flutterflow experience will be able to debug this since you can not actually show consistent code or rest calls being made.

Also do not know what kind of queries the get_campaign_details_dashboard is doing so it might as well be a totally different table involved that has a wrong policy set for it.

**Reply by sanjaychristian** [reply]
Thank you.  I will try to get help from flutterflow.

**Reply by sanjaychristian** [reply, marked as answer]
This is working now.  The problem was that you need to pass JWT token through Flutterflow.   In the header, make sure you spell the "Authorization" correctly (I had misspelled this) and create a variable something like the following:  Authorization: Bearer [variableName].  Also, declare this variable at API level so that you can pass the JWT token from Supabase.  This token can be retrieved from "Authenticated User" object.  Once you pass this to API, the auth.uid() value will have correct userid and you can do RLS on that.  Thank you all for helping!

**Comment by Matignaciom** [comment]
Hello! I am working on my project with flutterflow and something similar happens to me, please tell me if you have the same or a similar error, what happens to me is that the api test works perfectly but at the time of testing my App in the emulator or apk in only the first user who creates in the database works the response of the call api but in the other users or new users who I'm creating the API Call doesn't receive a response, I'm with FlutterFlow and SupaBase.

Here's my post:

https://github.com/orgs/supabase/discussions/29008
